src/main/java/org/folio/consortia/service/impl/TenantServiceImpl.java

```diff
@@ -81,15 +87,42 @@ public Tenant save(UUID consortiumId, Tenant tenantDto) {
     }
 
     checkTenantNotExistsAndConsortiumExistsOrThrow(consortiumId, tenantDto.getId());
-    Tenant savedTenant = saveTenant(consortiumId, tenantDto);
+    TenantEntity savedTenantEntity = saveTenantEntity(consortiumId, tenantDto);
+    var savedTenant = converter.convert(savedTenantEntity, Tenant.class);
 
     runInFolioContext(createFolioExecutionContextForTenant(tenantDto.getId(), currentTenantContext, folioMetadata),
-      () -> configurationClient.saveConfiguration(createConsortiaConfigurationBody(centralTenantId)));
+        () -> {
+          configurationClient.saveConfiguration(createConsortiaConfigurationBody(centralTenantId));
+          createPrimaryUserAffiliationsAsync(consortiumId, toEntity(consortiumId, tenantDto), tenantDto);
+        });
     log.info("save:: saved consortia configuration with centralTenantId={} by tenantId={} context", centralTenantId, tenantDto.getId());
 
     return savedTenant;
   }
 
+  public CompletableFuture<Void> createPrimaryUserAffiliationsAsync(UUID consortiumId, TenantEntity consortiaTenant, Tenant tenantDto) {
+    return CompletableFuture
+      .runAsync(() -> {
+        log.info("Start creating user primary affiliation for tenant {}", tenantDto.getId());
+        var users = usersClient.getUserCollection("", 0, Integer.MAX_VALUE);
```

Comment:
I am not sure how many users existing tenants can contain, did you think to use batch requests?

Reply:
yes. For sure It depends on amount of existing users. I beleive we can collect several thousands users in a single request without having any issues

Reply:
on bugfest we have 70k..., it's not real case, but still